man/rgblink.5

```diff
@@ -56,51 +50,109 @@ and
+must be a string.
+.It Specifying the active bank
+.Tg region
+The active bank can be set by specifying its type (memory region) and number.
+The possible types are:
+.Ic ROM0 , ROMX , VRAM , SRAM , WRAM0 , WRAMX , OAM ,
+and
+.Ic HRAM .
+However, if the type only contains a single bank, then the number can be omitted.
+The applicable types are:
+.Ic ROM0 ,
+.Ic ROMX No if Fl t No is passed to Xr rgblink 1 ,
+.Ic VRAM No if Fl d No is passed to Xr rgblink 1 ,
+.Ic WRAM0 ,
+.Ic WRAMX No if Fl w No is passed to Xr rgblink 1 ,
+.Ic OAM ,
+and
+.Ic HRAM .
+.Pq Ic SRAM No is the only type that can never have its bank number omitted.
 .Pp
-When a new bank statement is found, sections found after it will be placed right from the beginning of that bank.
-If the linker script switches to a different bank and then comes back to a previous one, it will continue from the last address that was used.
+After a bank specification, the
+.Dq current address
+is set to the last value it had for that bank.
+If the bank has never been active thus far, the
+.Dq current address
+defaults to the beginning of the bank
+.Pq e.g. Ad $4000 No for Ic ROMX No sections .
+.It Changing the current address
+A bank must be active for any of these directives to be used.
 .Pp
-The only three directives are
-.Ic ORG ,
-.Ic ALIGN ,
-and
-.Ic DS :
-.Bl -bullet
-.It
-.Ic ORG Ar addr
-sets the address in which new sections will be placed to
+.Ql Ic ORG Ar addr
+sets the
+.Dq current address
+to
 .Ar addr .
-It can not be lower than the current address.
-.It
-.Ic ALIGN Ar addr
-or
-.Ic ALIGN Ar addr , Ar offset
-will increase the address until it is aligned to the specified boundary
-.Po it tries to set to
+The new address cannot be less than the old one.
+.Pp
+.Ql Ic FLOATING
+causes all sections between it and the next
+.Ic ORG
+or bank specification to be placed at addresses automatically determined by
+.Nm .
+.Pp
+.Ql Ic ALIGN Ar addr , Ar offset
+increases the
+.Dq current address
+until it is aligned to the specified boundary (i.e. the
+.Ar align
+lowest bits of the address are equal to
+.Ar offset ) .
+If
 .Ar offset
-the number of bits specified by
-.Ar align :
-for example,
+is omitted, it is implied to be 0.
+For example, if the
+.Dq current address
+is $007,
 .Ql ALIGN 8
-will align to $100 ,
-and
+would set it to $100, and
 .Ql ALIGN 8 , 10
-will align to $10A
-.Pc .
-.It
-.Ic DS
-will increase the address by the specified non-negative amount.
+would set it to $00A.
+.Pp
+.Ql Ic DS Ar size
+increases the
+.Dq current address
+by
+.Ar size .
 .El
+.Ss Section placement
+A section can be placed simply by naming it (with a string).
+Its bank is set to the active bank, and its address to the
+.Dq current address .
+Any properties the section already possesses (whether from the linker script or the object files being linked) must be consistent with what the linker script specifies: the section's type must match, the section's bank number (if set) must match the active bank, etc.
```

Comment:
You should elaborate on the effect of alignment constraints here. The behavior is non-intuitive. (I don't even know what the actual behavior _is_.)

Reply:
"the current address must respect the section's alignment if set"?

Reply:
`ALIGN` in a linker script behaves like `DS ALIGN` in asm. (...Hm, since we have `DS` in linker scripts too, maybe this should be changed, for consistency and least surprise?

Reply:
Oh, I see what you're saying: `section align` in asm, and no alignment in the linker script. Then the linker script would place the section according to its own rules, and error if that placement contradicts the asm. Same with every other combo of asm and linker script placement directives. (e.g. if the linker script says `org $4567` and the asm says `romx[$4444]`, or `romx, align[8]`.)

Reply:
Yes, that's what I'm trying to convey. Perhaps "from earlier in the linker script or as read from the object files being linked"?

Reply:
"If a section has an alignment constraint, the address at which it is placed by the linker script must obey that constraint; otherwise, an error will occur."